### ISSUE TITLE
Is there anyway to delay computing of loss?

### ISSUE THREAD

**Issue by cloudhan** [apache/incubator-mxnet#5774]
Say, I have a compute graph,
``` python
pred = <blah blah graph> # the predicted output from graph
delayed_label = mx.sym.Variable(name="delayed_label")
loss = mx.sym.MakeLoss(<blah blah expression of symbol pred and symbol delayed_label>)
loss_output = mx.sym.Group([mx.sym.BlockGrad(pred), loss])
```
where `delayed_label` can only be retrieved after the `pred` is got. In this setting, seems I can do
```
exe = loss_output.simple_bind(...)
... # copy input to input array
exe.forward(is_train=False)
... # get pred output from exe.outputs[0] and then retrieve data for delayed_label
... # copy retrieved data to delayed_label array
exe.forward(is_train=True)
exe.backward()
... # update weights
```

Is there anyway I can do only one forward pass in this scenario instead of two?

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
